src/main/java/bio/terra/cloudfiletodatastore/deltalayer/DeltaLayerFileUploadedMessageProcessor.java

```diff
@@ -0,0 +1,76 @@
+package bio.terra.cloudfiletodatastore.deltalayer;
+
+import bio.terra.cloudfiletodatastore.*;
+import bio.terra.cloudfiletodatastore.deltalayer.model.json.PointCorrectionDestination;
+import bio.terra.cloudfiletodatastore.deltalayer.model.json.PointCorrectionRequest;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.InsertAllRequest;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * Fetches the json file referenced in the {@link FileUploadedMessage} from cloud storage, parses
+ * that file into Java using Gson, inserts the requisite data into BigQuery
+ */
+public class DeltaLayerFileUploadedMessageProcessor extends MessageProcessor {
+
+  private static final String EXPECTED_CONTENT_TYPE = "application/json";
+
+  private final ResourceFetcher resourceFetcher;
+
+  private BigQuery bqForTest;
+
+  private static final Logger logger =
+      Logger.getLogger(DeltaLayerFileUploadedMessageProcessor.class.getName());
+
+  public DeltaLayerFileUploadedMessageProcessor(FileUploadedMessage message) {
+    super(message);
+    this.resourceFetcher = new GcsFileFetcher(message.getSourceBucket(), message.getResourceName());
+  }
+
+  @VisibleForTesting
+  DeltaLayerFileUploadedMessageProcessor(
+      FileUploadedMessage message, ResourceFetcher rf, BigQuery bq) {
+    super(message);
+    this.resourceFetcher = rf;
+    this.bqForTest = bq;
+  }
+
+  @Override
+  public void processMessage() {
+    if (!EXPECTED_CONTENT_TYPE.equals(message.getContentType())) {
+      logger.warning(
+          String.format(
+              "Unexpected content type %s, concluding processing early", message.getContentType()));
+      return;
+    }
```

Comment:
Fair.  The content-type is only useful as long as it's in sync with the actual content and it's probably not a huge deal to fail processing with a JSON parsing error rather than skipping out upon a mismatched content-type.